lib/ansible/plugins/inventory/kubevirt.py

```diff
@@ -0,0 +1,219 @@
+              description:
+              - In case of multiple network attached to virtual machine, define which interface should be returned as primary IP
+                address.
+              type: str
+          api_version:
+              description:
+              - "Specify the KubeVirt API version."
+              type: str
+    requirements:
+    - "openshift >= 0.6"
+    - "PyYAML >= 3.11"
+'''
+
+EXAMPLES = '''
+# File must be named kubevirt.yaml or kubevirt.yml
+
+# Authenticate with token, and return all virtual machines for all namespaces
+plugin: kubevirt
+connections:
+ - host: https://kubevirt.io
+   token: xxxxxxxxxxxxxxxx
+   ssl_verify: false
+
+# Use default config (~/.kube/config) file and active context, and return vms with interfaces
+# connected to network myovsnetwork and from namespace vms
+plugin: kubevirt
+connections:
+  - namespaces:
+      - vms
+    network_name: myovsnetwork
+'''
+
+from ansible.plugins.inventory.k8s import K8sInventoryException, InventoryModule as K8sInventoryModule, format_dynamic_api_exc
+
+try:
+    from openshift.dynamic.exceptions import DynamicApiError
+except ImportError:
+    pass
+
+
+API_VERSION = 'kubevirt.io/v1alpha3'
+
+
+class InventoryModule(K8sInventoryModule):
+    NAME = 'kubevirt'
+
+    def fetch_objects(self, connections):
+        client = self.get_api_client()
+
+        if connections:
+            for connection in connections:
+                client = self.get_api_client(**connection)
+                name = connection.get('name', self.get_default_host_name(client.configuration.host))
+                if connection.get('namespaces'):
+                    namespaces = connection['namespaces']
+                else:
+                    namespaces = self.get_available_namespaces(client)
+                interface_name = connection.get('network_name')
+                api_version = connection.get('api_version', API_VERSION)
+                for namespace in namespaces:
+                    self.get_vms_for_namespace(client, name, namespace, interface_name, api_version)
+        else:
+            name = self.get_default_host_name(client.configuration.host)
+            namespaces = self.get_available_namespaces(client)
+            for namespace in namespaces:
+                self.get_vms_for_namespace(client, name, namespace, None, api_version)
+
+    def get_vms_for_namespace(self, client, name, namespace, interface_name=None, api_version=None):
+        v1_vm = client.resources.get(api_version=api_version, kind='VirtualMachineInstance')
+        try:
+            obj = v1_vm.get(namespace=namespace)
+        except DynamicApiError as exc:
+            self.display.debug(exc)
+            raise K8sInventoryException('Error fetching Virtual Machines list: %s' % format_dynamic_api_exc(exc))
+
+        namespace_group = 'namespace_{0}'.format(namespace)
+        namespace_vms_group = '{0}_vms'.format(namespace_group)
+
+        self.inventory.add_group(name)
```

Comment:
`add_group` returns a sanitized name, to ensure the `add_child` below doesn't error out either use that name or sanitize `name` beforehand. You can use `self._sanitize_group_name()`

Reply:
done